job_templates/cyclic_cc_pt/config_fed_client.conf

```diff
@@ -0,0 +1,106 @@
+  app_config = ""
+
+  # path to defined PyTorch network
+  model_class_path = "net.Net"
+
+  # Client Computing Executors.
+  executors = [
+    {
+      # tasks the executors are defined to handle
+      tasks = ["train"]
+
+      # This particular executor
+      executor {
+
+        # Eexecutor name : PTClientAPILauncherExecutor
+        # This is an executor for pytorch + Client API. The underline data exchange is using FilePipe.
+        path = "nvflare.app_opt.pt.client_api_launcher_executor.PTClientAPILauncherExecutor"
+
+        args {
+
+          # This executor take an component named "launcher"
+          launcher_id = "launcher"
+
+          # Timeout in seconds for waiting for a heartbeat from the training script. Defaults to 30 seconds.
+          # Please refer to the class docstring for all available arguments
+          heartbeat_timeout = 60
+
+          # data_exchange_path: is the directory location of the parameters exchange.
+          # If empty string, it will be set to the app_dir of the running job.
+          # You can also set it to an absolute path in your system.
+          data_exchange_path = ""
+
+          # format of the exchange parameters
+          params_exchange_format =  "pytorch"
+
+          # if the transfer_type is FULL, then it will be sent directly
+          # if the transfer_type is DIFF, then we will calculate the
+          # difference VS received parameters and send the difference
+          params_transfer_type = "DIFF"
+
+          # if train_with_evaluation is true, the executor will expect
+          # the custom code need to send back both the trained parameters and the evaluation metric
+          # otherwise only trained parameters are expected
+          train_with_evaluation = true
+
+          # if launch_once is true, the executor will only call launcher.launch_task once
+          # for the whole job, if launch_once is false, the executor will call launcher.launch_task
+          # everytime it receives a task from server
+          launch_once = true
+        }
+      }
+    }
+    {
+      # All tasks prefixed with cyclic_ are routed to CyclicClientController
+      tasks = ["cyclic_*"]
+      executor {
+        # client-side controller for cyclic workflow
+        path = "nvflare.app_common.ccwf.CyclicClientController"
+        args {
+          # maps to the train task of the PTClientAPILauncherExecutor
+          learn_task_name = "train"
+          persistor_id = "persistor"
+          shareable_generator_id = "shareable_generator"
+        }
+      }
+    }
+  ]
+
+  # this defined an array of task data filters. If provided, it will control the data from server controller to client executor
+  task_data_filters =  []
+
+  # this defined an array of task result filters. If provided, it will control the result from client executor to server controller
+  task_result_filters = []
```

Comment:
This description is not 100% accurate in CCWF.

The task_data_filters filters things coming from Client side Controller -> Client side Executor right?
The task_result_filters filters things sending from Client side Executor -> Client side Controller?

@yhwen can you add more context in this?
We introduce the "direction" as well right?

Reply:
Oh right good catch, I forgot to update these comments from the previous template

Reply:
Updated comment for ccwf and filter direction (in, out, inout)